pkg/manifest/manifest.go

```diff
@@ -115,6 +158,34 @@ func ParseManifests(r io.Reader) ([]Manifest, error) {
 		if len(m.Raw) == 0 || bytes.Equal(m.Raw, []byte("null")) {
 			continue
 		}
+		if err := addIfNotDuplicateResource(m, theseIds); err != nil {
+			return manifests, err
+		}
 		manifests = append(manifests, m)
 	}
 }
+
+// validateResourceId ensures the id contains the required fields per
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields.
+func validateResourceId(id ResourceId) error {
+	if id.Kind == "" || id.Name == "" {
+		return fmt.Errorf("Resource with fields %s must contain kubernetes required fields kind and name", id)
+	}
+	return nil
+}
+
+func truncateVersion(apiVersion string) string {
```

Comment:
You can drop this, now that you've removed its consumers.

Reply:
yep